### PR TITLE
Detect html (see #75)

### PR DESCRIPTION
**Pull request by mozfreddyb**
Having looked at the TokenSink example in #75, this seemed mostly straightfoward.
I found a couple of other nits along my way that I figured I might just pick up and do within this pull request, but I'm happy to drop them or turn them into individual PRs, if I must.

### CODE REVIEW

**Comment by mozfreddyb** [comment]
Not really sure if and how the breaking test run (compiling `memchr` on rust 1.36) affects this and where to go from here. Looking forward to more feedback.

**Comment by notriddle** [comment]
It's fine. It just means we bump to 1.41

File: src/lib.rs [hunk 1]
```diff
@@ -155,6 +157,55 @@ pub fn clean_text(src: &str) -> String {
...
+        SanitizationTokenizer {
+            was_sanitized: false,
+        }
+    }
+}
+
+impl TokenSink for SanitizationTokenizer {
+    type Handle = ();
+    fn process_token(&mut self, token: Token, _line_number: u64) -> TokenSinkResult<()> {
+        println!("Heh, token! {:?}", token);
```
Comment (notriddle) [inline src/lib.rs:197]: This probably shouldn't be here. It's for your debugging purposes, right?

**Comment by notriddle** [comment]
Other than the println statement that you accidentally left in, this code seems fine.

File: src/lib.rs (hunk 1 above)
Comment (mozfreddyb) [reply src/lib.rs]: oops. well spotted.

**Comment by mozfreddyb** [comment]
I hope the tests are along the lines of what you thought. Before closing #75, do you still want a `Builder::empty()` that allows nothing?

**Review by notriddle — APPROVED** [review]
bors r+

**Pull request merged**